lib/src/line_splitting/solve_state.dart

```diff
@@ -197,10 +194,10 @@ class SolveState {
         for (var value = 1; value < rule.numValues; value++) {
           var boundRules = unsplitRules.clone();
 
-          List<Rule> mustSplitRules;
+          List<Rule>? mustSplitRules;
```

Comment:
Personally I would do:
```dart
var mustSplitRules = <Rule>[];
var valid = boundRules.tryBind(_splitter.rules, rule, value, mustSplitRules.add);
```
And then change the `if (mustSplitRules != null)` block below to `if (mustSplitRules.isNotEmpty)`.

It's an extra allocation but I suspect the readability improvement will outweigh the performance cost.